### DISCUSSION TITLE
Amazon Bedrock model support

### DISCUSSION THREAD

**Discussion by advait404** [Taewan-P/gpt_mobile#50] [Ideas]
Hi, 
I think it would be nice if the app can support models from amaozn bedrock. 

Currently I have not been able to find a single android app that can do this, and I think it should not be a major challenge since calling the APIs is pretty straightforward.

Maybe initially only the anthropic models can be supporter, then it can be expanded to other models as well.

**Comment by Taewan-P** [comment]
Hello!
What are the benefits of using models from Amazon Bedrock? (I've never used one before) Is there something that points out compared to other models?

**Comment by advait404** [comment]
There are a few benefits of using aws bedrock.

- provides many text and image models in a centralised location. (Claude, llama 3, mistral, cohere, stable diffusion, etc)
- mostly standardised methods to share and receive responses from llms. Has support for images, steaming, etc too.
- pay as you go pricing, similar to most other API billing methods
- better security, especially since its all managed through the aws account
- additional features like guardrails, fine tuning, etc

Honestly I have not found a mobile app for bedrock, and I really wanted to use it outside the webui, which is pretty bad on mobile. So this can be a pretty unique feature.

I have worked on implementing this (in python) so I can somewhat help in how it can be implemented.

**Comment by Taewan-P** [comment]
I will consider this, thanks.
